### DISCUSSION TITLE
Documentation Request

### DISCUSSION THREAD

**Discussion by ngl5000** [tighten/ziggy#791] [General]
I'm using Laravel 11, InertiaJS SSR, Breeze Starter Kit, Vue3, and Ziggy. I'm also using persistent Vue layouts using defineOptions. 

I'm having trouble using Ziggy in SSR mode the way I was before transitioning my app to SSR. Specifically, I can't find a straightforward implementation to get route() available in my setup scripts. I would have expected better documentation around this as I haven't strayed much from the mainstream laravel stacks. I've found a lot of old dated conversations about setups in Mix, but nothing current and nothing in any official documentation.

Is there a good way to do this without being forever plagued by: ReferenceError: route is not defined

**Comment by bakerkretzmar** [comment]
Sorry, missed this. Not sure if it was in the Readme when you posted this but it's there now: https://github.com/tighten/ziggy?tab=readme-ov-file#vue.

Unfortunately the only way to make `route()` available inside `<script setup>` is to `inject()` it every time. Our Vue plugin `provide()`s it to make this possible, and also makes it available in templates.

**Reply by bakerkretzmar** [reply]
Actually are you using the `@routes` Blade directive? That also makes it available everywhere, you'll just have to add a TypeScript declaration somewhere to let your IDE know about it.
